python/ray/experimental/channel/serialization_context.py

```diff
@@ -9,10 +9,35 @@ class _SerializationContext:
     def __init__(self):
         self.use_external_transport: bool = False
         self.tensors: List["torch.Tensor"] = []
+        # Buffer for transferring data between tasks in the same worker process.
+        # The key is the channel ID, and the value is the data.
+        self.intra_process_channel_buffers: Dict[str, Any] = {}
+        # The number of readers for each channel. When the number of readers
+        # reaches 0, remove the data from the buffer.
+        self.channel_id_to_num_readers: Dict[str, int] = {}
 
     def set_use_external_transport(self, use_external_transport: bool) -> None:
         self.use_external_transport = use_external_transport
 
+    def set_data(self, channel_id: str, value: Any, num_readers: int) -> None:
+        assert channel_id not in self.intra_process_channel_buffers
+        assert channel_id not in self.channel_id_to_num_readers
+        self.intra_process_channel_buffers[channel_id] = value
```

Comment:
I don't use a lock when reading/writing the buffer because a DAG node actor will only execute one task at a time in `do_exec_tasks`. It will not execute multiple Ray tasks on a single actor simultaneously.

Reply:
Please add that context as a comment, thanks!

Reply:
Added https://github.com/ray-project/ray/pull/45591/commits/085bcf9054dee3f9752deac24148f7e9859f7ec0